### PR TITLE
Refactored password validation indicators in signup form

### PR DESCRIPTION
**Pull request by farzanashampa7**

### CODE REVIEW

File: src/components/FormElements/Password/supplement/Password.test.tsx
Comment (AsifOni) [inline src/components/FormElements/Password/supplement/Password.test.tsx:1]: There is no test to validate if emoji is passed, then field should throw error. Please add the test case

File: src/components/SignupForm/SignupForm.tsx [hunk 1]
```diff
@@ -259,87 +303,31 @@ const SignupForm = () => {
...
-                </p>
-              </div>
+              <PassWordValidationIndicator
+                contentKey="lowercase"
+                passwordValidationRegEx={passwordValidationRegEx}
+              />
+              <PassWordValidationIndicator
+                contentKey="numbers"
+                passwordValidationRegEx={passwordValidationRegEx}
+              />
```
Comment (AsifOni) [inline src/components/SignupForm/SignupForm.tsx]: In all component, same `passwordValidationRegEx` has been passed. Two approach can be taken here

1. Don't pass regEx as prop rather use `passwordValidationSchema` directly inside component OR
2. Pass only specific regEx

**Review by AsifOni — CHANGES_REQUESTED** [review]
It seems emoji is allowed. Please check the regEx, I think we need to block emojis to be set as password

image

File: src/components/FormElements/Password/supplement/Password.test.tsx
Comment (farzanashampa7) [reply src/components/FormElements/Password/supplement/Password.test.tsx:1]: Screenshot 2024-05-01 at 5 21 02 PM

vaiya I tested at my end and here emoji isn't allowed, not sure if I am missing something

Comment (farzanashampa7) [reply src/components/FormElements/Password/supplement/Password.test.tsx:1]: Vaiya do we want to block typing emojis or to throw an error when emoji is passed?

Comment (AsifOni) [reply src/components/FormElements/Password/supplement/Password.test.tsx:1]: We want to throw an error while emoji is typed

**Review by AsifOni — APPROVED** [review]

**Pull request merged**